core-plugins/src/main/java/co/cask/hydrator/plugin/common/KinesisUtil.java

```diff
@@ -0,0 +1,150 @@
+
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.model.CreateStreamRequest;
+import com.amazonaws.services.kinesis.model.DescribeStreamRequest;
+import com.amazonaws.services.kinesis.model.ResourceNotFoundException;
+import com.amazonaws.services.kinesis.model.StreamStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utility methods to help create and manage Kinesis Streams
+ */
+public final class KinesisUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KinesisUtil.class);
+  private static final long TIMEOUT = 3 * 60 * 1000;
+  private static final long SLEEP_INTERVAL = 1000 * 10;
+  private KinesisUtil() {
+  }
+
+  /**
+   * Creates an Amazon Kinesis stream if it does not exist and waits for it to become available
+   *
+   * @param kinesisClient The {@link AmazonKinesisClient} with Amazon Kinesis read and write privileges
+   * @param streamName The Amazon Kinesis stream name to create
+   * @param shardCount The shard count to create the stream with
+   * @throws IllegalStateException Invalid Amazon Kinesis stream state
+   * @throws IllegalStateException Stream does not become active before the timeout
+   */
+  public static void createAndWaitForStream(AmazonKinesisClient kinesisClient, String streamName, int shardCount) {
+    if (streamExists(kinesisClient, streamName)) {
+      StreamStatus streamStatus = getStreamState(kinesisClient, streamName);
+      switch (streamStatus) {
+        case DELETING:
+          long waitTimeDelete = System.currentTimeMillis() + TIMEOUT;
+          while (System.currentTimeMillis() < waitTimeDelete && streamExists(kinesisClient, streamName)) {
+            try {
+              LOG.debug("Deleting Stream {} ", streamName);
+              TimeUnit.MILLISECONDS.sleep(SLEEP_INTERVAL);
+            } catch (InterruptedException e) {
+              Thread.currentThread().interrupt();
+            }
+          }
+          if (streamExists(kinesisClient, streamName)) {
+            throw new IllegalStateException(String.format("Timed out waiting for stream {} to delete", streamName));
+          }
+          createStream(streamName, shardCount, kinesisClient);
+          break;
+        case ACTIVE:
+          LOG.info("Stream {} already exists", streamName);
+          return;
+        case CREATING:
+          LOG.info("Stream {} is being created", streamName);
+          break;
+        case UPDATING:
+          LOG.info("Stream {} is being updated", streamName);
+          return;
+        default:
+          throw new IllegalStateException(String.format("Illegal stream state: {}", streamStatus));
+      }
+    } else {
+      createStream(streamName, shardCount, kinesisClient);
+    }
+    long waitTimeCreate = System.currentTimeMillis() + TIMEOUT;
+    while (System.currentTimeMillis() < waitTimeCreate) {
```

Comment:
both create and delete eventually call ``getStreamState`` as the exit condition from the loop.